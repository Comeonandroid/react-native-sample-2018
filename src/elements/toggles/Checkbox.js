import React from 'react'
import { MKCheckbox } from 'react-native-material-kit'
import styled, { withTheme } from 'styled-components'
import { PropTypes } from 'prop-types'

import { View, Text } from '../index'

const Checkbox = ({ checked, onCheckedChange, theme, children, inset, ...rest }) => (
  <Wrapper>
    <Input
      checked={checked}
      fillColor={theme.colors.toggleFill}
      inset={inset}
      borderOnColor={theme.colors.inputBorder}
      borderOffColor={theme.colors.inputBorder}
      rippleColor={`rgba(250,166,50,.15)`}
      onCheckedChange={onCheckedChange}
      {...rest}
    />
    { children && <Label>{ children }</Label> }
  </Wrapper>
)

const Wrapper = styled(View) `
  flex-direction: row;
  align-items: center;
`

const Input = styled(MKCheckbox) `
  borderWidth: 1;
  background-color: ${ ({ theme }) => theme.colors.inputBackground };
`

const Label = styled(Text) `
  color: ${ ({ theme }) => theme.colors.font };
  font-weight: 400;
  font-size: 10px;
  margin-left: 6px;
`

Checkbox.propTypes = {
  children: PropTypes.node,
  checked: PropTypes.bool,
  inset: PropTypes.number,
  onCheckedChange: PropTypes.func,
  theme: PropTypes.object
}

Checkbox.defaultProps = {
  checked: false,
  inset: 3
}

export default withTheme(Checkbox)