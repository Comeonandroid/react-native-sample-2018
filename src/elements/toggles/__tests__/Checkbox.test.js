import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'

import Checkbox from '../Checkbox'

const setup = propOverrides => {
  const props = Object.assign({
    children: null,
    checked: false,
    onCheckedChange: jest.fn(),
  }, propOverrides)

  const component = <Checkbox {...props} />
  const wrapper = shallow(component)

  return {
    props,
    wrapper,
    component
  }
}

describe('<Checkbox />', () => {
  // react-test-renderer dosen't work with MKCheckbox
  it('renders correctly', () => {
    const { wrapper } = setup()
    expect(wrapper).toMatchSnapshot()
  })


  it('renders correctly with children', () => {
    const { props, wrapper } = setup({ children: 'Test' })

    expect(wrapper.children().last().text()).toEqual(props.children)
  })
})
