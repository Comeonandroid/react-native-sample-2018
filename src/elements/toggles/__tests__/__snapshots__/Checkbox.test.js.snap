// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> renders correctly 1`] = `
<Checkbox
  checked={false}
  inset={3}
  onCheckedChange={[MockFunction]}
/>
`;
